src/dmd/expressionsem.d

```diff
@@ -7626,6 +7626,13 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
         case Tsarray:
             {
+                if (exp.e2.type.toBasetype().ty == Tbool)
+                {
+                    /* Protect against bools initialized with void or unions
```

Comment:
Is there a test for the `void` case?

Reply:
No way to make a reliable one. That's why I did the reproducible union test.